lib/ModelMixins/AutoRefreshingMixin.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  computed,
+  IReactionDisposer,
+  onBecomeObserved,
+  onBecomeUnobserved,
+  reaction
+} from "mobx";
+import { now } from "mobx-utils";
+import Constructor from "../Core/Constructor";
+import Model from "../Models/Model";
+import MappableTraits from "../Traits/MappableTraits";
+
+type AutoRefreshing = Model<MappableTraits>;
+
+export default function AutoRefreshingMixin<
+  T extends Constructor<AutoRefreshing>
+>(Base: T) {
+  abstract class AutoRefreshingMixin extends Base {
+    _autoRefreshDisposer: IReactionDisposer | undefined;
+
+    /* Return the interval in seconds to poll for updates. */
+    abstract refreshInterval: number | undefined;
+
+    /* Call hook for refreshing the item */
+    abstract refreshData(): void;
+
+    constructor(...args: any[]) {
+      super(...args);
+      // We should only poll when our map items have consumers
+      onBecomeObserved(this, "mapItems", () => {
+        this._autoRefreshDisposer = reaction(
+          () => this._pollingTimer,
+          () => {
+            this.refreshData();
+          }
+        );
+      });
+      onBecomeUnobserved(this, "mapItems", () => {
+        if (
+          this._autoRefreshDisposer !== undefined &&
+          this._autoRefreshDisposer !== null
+        ) {
+          this._autoRefreshDisposer();
+        }
+      });
+    }
+
+    @computed
+    protected get _pollingTimer(): number | undefined {
```

Comment:
Can this be private? It looks like it can be. That'd be better because the name is much less important for a private field than for a protected one. And the name isn't quite right. It's not a timer, it's a time. But it's a funny time that only updates every refresh interval. So, hard to name, better if we don't have to sweat it too much.